components/org.wso2.carbon.identity.api.server.api.resource/org.wso2.carbon.identity.api.server.api.resource.v1/src/main/java/org/wso2/carbon/identity/api/server/api/resource/v1/core/ServerApiResourceManagementService.java

```diff
@@ -0,0 +1,473 @@
+import org.wso2.carbon.identity.application.common.model.ApiResource;
+import org.wso2.carbon.identity.application.common.model.Scope;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.identity.api.server.api.resource.v1.constants.ApiResourceMgtEndpointConstants.ASC_SORT_ORDER;
+import static org.wso2.carbon.identity.api.server.api.resource.v1.constants.ApiResourceMgtEndpointConstants.DEFAULT_LIMIT;
+import static org.wso2.carbon.identity.api.server.api.resource.v1.constants.ApiResourceMgtEndpointConstants.DESC_SORT_ORDER;
+import static org.wso2.carbon.identity.api.server.common.Constants.V1_API_PATH_COMPONENT;
+
+/**
+ * Server Api Resource Management Service.
+ */
+public class ServerApiResourceManagementService {
+
+    private static final ServerApiResourceManagementService instance = new ServerApiResourceManagementService();
+
+    private static final Log log = LogFactory.getLog(ServerApiResourceManagementService.class);
+
+    private ServerApiResourceManagementService() {
+
+    }
+
+    public static ServerApiResourceManagementService getInstance() {
+
+        return instance;
+    }
+
+    /**
+     * Add Api resource.
+     *
+     * @param apIResourceCreationModel Api resource creation model.
+     * @return Response.
+     */
+    public ApiResourceResponse addApiResourceWithResourceId(ApiResourceCreationModel apIResourceCreationModel) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Adding Api resource with resource id: " + apIResourceCreationModel.getIdentifier());
+        }
+        ApiResource createdApiResource;
+        try {
+            ApiResource apiResource = createApiResource(apIResourceCreationModel);
+            createdApiResource = ApiResourceManagementServiceHolder.getApiResourceManager()
+                    .addApiResource(apiResource, ContextLoader.getTenantDomainFromContext());
+
+            if (createdApiResource == null) {
+                log.error(ErrorMessage.ERROR_CODE_ADD_API_RESOURCE.getDescription());
+                throw ApiResourceManagementEndpointUtil.handleException(Response.Status.INTERNAL_SERVER_ERROR,
+                        ErrorMessage.ERROR_CODE_ADD_API_RESOURCE);
+            }
+        } catch (ApiResourceManagementException e) {
+            throw ApiResourceManagementEndpointUtil.handleApiResourceMgtException(e);
+        }
+        return buildApiResourceResponse(createdApiResource);
+    }
+
+    /**
+     * Build ApiResourceResponse from ApiResource.
+     *
+     * @param apiResource ApiResource object.
+     * @return ApiResourceResponse object.
+     */
+    private ApiResourceResponse buildApiResourceResponse(ApiResource apiResource) {
```

Comment:
Shall we organize the public methods to the top and private methods to the end?

Reply:
fixed with https://github.com/wso2/identity-api-server/pull/473/commits/907f522f5b71eb350e9aa926b5ba3580f0022746